File: study-cpp-demo/Student.cpp
#include "Student.h"
#include "iostream"
using namespace std;

Student::Student() {
	cout << "Student constuct"<< endl;
	Student::age = 0;
	Student::name = 'a';
};
Student::Student(int age) {
	Student::age = age;
	Student::name = 'b';
};

Student::Student(int a,char b) {
	age = a;
	name = b;
};
Student::~Student() {
	cout << "Student destruct"<<endl;
};
void Student::setAge(int age) {
	Student::age = age;
};
void Student::setName(char name) {
	Student::name = name;
};
int Student::getAge() {
	return age;
};
char Student::getName() {
	return name;
};
void Student::printStudent() {
	cout << "age value :" << age << "\nname value:" << name << endl;
};


static void shows() {
	cout << "shows method" << endl;
}
void Student::print() {
	shows();
	cout << "static method" << endl;
};

File: any-code/src/com/copy/Student.java
package com.copy;

/**
 * 定义一个引用数据类型对象
 *
 * 
 */
public class Student implements Cloneable{

    /**
     * 学生姓名
     */
    private String name;

    /**
     * 学生年龄
     */
    private int age;

    public Student() {

    }

    public Student(String name, int age) {
        this.name = name;
        this.age = age;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Override
    public String toString() {
        return "Student{" +
                "name='" + name + '\'' +
                ", age=" + age +
                '}';
    }


    @Override
    public Object clone() throws CloneNotSupportedException {
        return super.clone();
    }
}
